docs/Developing the HMI UI/index.md

```diff
@@ -0,0 +1,411 @@
+
+The `UI.Show` request has a parameter `templateConfiguration` which includes a string for the requested layout.
+
+The `UI.SetDisplayLayout` request includes a string parameter, `displayLayout`.
+
+`UI.Show` is meant to be the preferred method because a `UI.Show` request can change the layout of the screen and the screen contents in a single request. This helps prevent lag and screen flashing when an app wants to change an app template. 
+
+An SDL app should only request to view templates that are supported in the HMI Capabilities. The HMI may return a failed response to Core in the event an unsupported template is requested. 
+
+More on HMI capabilities will be discussed later in this guide.
+
+## Supported Template Views
+
+A reference list for all supported template views can be found [here](https://smartdevicelink.com/en/guides/sdl-overview-guides/user-interface/supported-templates/).
+
+This list shows screen shots of the 15 supported template views and how their text, graphic, and soft button components are layed out.
+
+The defined strings for each template can be found in `PredefinedLayout` struct in the Mobile API RPC Specification. 
+
+## Creating the App Menu
+
+The in App Menu should be made accessible from an app's template view. Please note the example placement of the hamburger menu icon in the top right of this screen shot.
+
+![Menu Icon](./assets/menu_circled.jpg)
+
+!!! NOTE
+If the user chooses to open the menu, the HMI must send a `UI.OnSystemContext` notification with the `SystemContext` enum: `MENU`. After the user exits the menu, another `UI.OnSystemContext` notification must be sent with the `SystemContext` value: `MAIN`.
+!!!
+
+The contents of the HMI's menu are populated by the RPC `UI.AddCommand`. Each `UI.AddCommand` received corresponds to an individual menu item. When the user selects a menu item, the HMI should send a `UI.OnCommand` notification. We believe it is best practice to exit the menu after a user makes a selection from the list of commands.
+
+There is some minor customization for the app menu. An HMI can choose to implement the app menu in a tile view, list view, or both. If an app has a preference for a type of menu layout, the HMI will receive a `UI.SetGlobalProperties` request from SDL Core.
+
+![List Menu](./assets/list_menu.png)
+
+![Tiles Menu](./assets/tiles_menu.png)
+
+## Implementing Popups
+
+There are several rpcs that are meant to display a popup or an overlay to the user.
+
+
+### `UI.Alert`
+
+Alert is a simple popup that can contain an image, text, and buttons.
+
+ ![Alert](./assets/alert.png)
+
+### `UI.PerformInteraction`
+
+`PerformInteraction` is a ppoup with contents displayed similar to the app menu
+
+![Perform Interaction](./assets/perform_interaction.png)
+
+### `UI.Slider`
+
+`Slider` is a popup that allows the user to enter an input via a slider button.
+
+![Slider](./assets/slider.png)
+
+### `UI.ScrollableMessage`
+
+![Scrollable Message](./assets/scrollable_message.png)
```

Comment:
Should the image be after the description, as is the case with the rest of the RPCs?